### PR TITLE
Use prettier

### PR DESCRIPTION
**Pull request by codingfabi**
see issue :#396

tl;dr:
Prettier in combination with pre-commit hooks and husky formats the code automatically with predefined optioins before your commits. this gurantees a generalized code style and prevents the lint-checker from failing. 

### Architectural/Backend Changes
- [x] install the prettier package
- [x] create a prettier config file to manage formatting options
- [x] install husky and configure a pre-commit hook
- [x] refactor the whole projects using those two packages

### CODE REVIEW

**Comment by codingfabi** [comment]
Feel free to discuss the defined formatting options and give me your opinion on the whole thing

**Comment by TheSlimvReal** [comment]
How do you get this hook executed? It did not work when I commited something in WebStorm.

**Comment by sleidig** [comment]
at least to some extent tslint offers the same thing, doesn't it? The prettier docs claim there still is a point to use both together (https://prettier.io/docs/en/integrating-with-linters.html and there are integration plugins) but I am not fully sure. Do you have experience with both, @codingfabi ?

Definitely a good idea to include this in pre-commit hooks!

**Comment by codingfabi** [comment]
> 
> 
> How do you get this hook executed? It did not work when I commited something in WebStorm.

In VSCode, it worked on my machine automatically with the commit. Did you npm install? 
It could also be a problem with access authorization, because prettier needs a CL run as administrator. 
maybe that is your problem here, try to run _prettier --write '/**/*.js'_  in your command line in webstorm and give me the output

**Comment by codingfabi** [comment]
> 
> 
> at least to some extent tslint offers the same thing, doesn't it? The prettier docs claim there still is a point to use both together (https://prettier.io/docs/en/integrating-with-linters.html and there are integration plugins) but I am not fully sure (`tslint --fix` can also auto-format your code). Do you have experience with both, @codingfabi ?
> 
> Definitely a good idea to include this in pre-commit hooks!

Yes I have a project at work where we use both. We could also consider to add jsut tslint --fix as a pre-commit hook. But I like the easy configuration from prettier without touching the tslint highlighting. But that is a personal preference. What do you think?

**Comment by sleidig** [comment]
I am slightly afraid of maintaining similar things in two different places. But if you have positive experience with that setup let's just do that. 
Should we use one of the integration packages to sync rules from tslint to the new prettier config then? ( https://prettier.io/docs/en/integrating-with-linters.html) I am happy with the current tslint config but also open to adaptions.

File: src/app/child-dev-project/aser/model/aser.ts [hunk 1]
```diff
@@ -22,21 +22,8 @@ import { DatabaseEntity } from '../../../core/entity/database-entity.decorator';
...
-    'Read Paragraph',
-  ];
-  static MathLevels = [
-    'Nothing',
-    'Numbers 1-9',
-    'Numbers 10-99',
-    'Subtraction',
-    'Division',
-  ];
+  static ReadingLevels = ['Nothing', 'Read Letters', 'Read Words', 'Read Sentence', 'Read Paragraph'];
```
Comment (sleidig) [inline src/app/child-dev-project/aser/model/aser.ts]: is there a way to force consistent linebreaks (all or nothing) but leave the choice between breaks or not to the user? (eslint seems to have that option)

File: src/app/child-dev-project/attendance/child-attendance/child-attendance.component.spec.ts [hunk 2]
```diff
@@ -156,33 +155,30 @@ const ATTENDANCE_ENTITIES: AttendanceMonth[] = (() => {
   return data;
 })();
 
-
```
Comment (sleidig) [inline src/app/child-dev-project/attendance/child-attendance/child-attendance.component.spec.ts]: I do like the opportunity to add 2-3 empty lines separating different logical blocks

File: src/app/child-dev-project/aser/model/aser.ts (hunk 1 above)
Comment (sleidig) [reply src/app/child-dev-project/aser/model/aser.ts]: apparently not possible with prettier ...

**Pull request merged**